examples/wallet/test/components/SendTransaction.spec.js

```diff
@@ -0,0 +1,434 @@
+          expect(getAddressInput(component).prop('isInvalid')).toBeFalsy();
+        });
+
+        describe('WHEN address is entered', () => {
+          beforeAll(async () => {
+            getAddressInput(component).simulate('change', {
+              target: { value: mockedAddress }
+            });
+            await new Promise(resolve =>
+              setTimeout(resolve, config.get('formDebounceInterval'))
+            );
+          });
+          test('THEN the value changed', () => {
+            expect(getAddressInput(component).prop('value')).toBe(
+              mockedAddress
+            );
+          });
+          test('AND the address is not valid', () => {
+            expect(getAddressInput(component).prop('isValid')).toBeFalsy();
+          });
+          test('AND the address is invalid', () => {
+            expect(getAddressInput(component).prop('isInvalid')).toBeTruthy();
+          });
+          // FIXME: the children'll always be passed, I have to find a way to check if it's being displayed. isEmptyRender() doesn't work because the feedback is implemented as a forwardRef, the form doesn't know if it has to render.
+          test('AND an error message is displayed', () => {
+            expect(getAddressFeedback(component).prop('children')).toMatch(
+              'Please enter a valid address'
+            );
+          });
+        });
+      });
+
+      describe('GIVEN the (mocked) chain libs report an address as valid', () => {
+        let component;
+        let isValidAddress;
+        beforeAll(() => {
+          ({ component, isValidAddress } = setup(100, defaultNodeSettings));
+          isValidAddress.resolves(true);
+        });
+        test('AND the address is not valid or invalid', () => {
+          expect(getAddressInput(component).prop('isValid')).toBeFalsy();
+          expect(getAddressInput(component).prop('isInvalid')).toBeFalsy();
+        });
+
+        describe('WHEN address is entered', () => {
+          beforeAll(async () => {
+            getAddressInput(component).simulate('change', {
+              target: { value: mockedAddress }
+            });
+            // validator is debounced, we have to wait for it.
+            await new Promise(resolve =>
+              setTimeout(resolve, config.get('formDebounceInterval'))
+            );
+          });
+          test('THEN the value changed', () => {
+            expect(getAddressInput(component).prop('value')).toBe(
+              mockedAddress
+            );
+          });
+          test('AND the address is valid', () => {
+            expect(getAddressInput(component).prop('isValid')).toBeTruthy();
+          });
+          test('AND the address is not invalid', () => {
+            expect(getAddressInput(component).prop('isInvalid')).toBeFalsy();
+          });
+          // TODO: check no error message is displayed
+        });
+      });
+    });
+  });
+});
```

Comment:
A very interesting test, but it looks very expensive to develop. It would be necessary to define criteria to select which components should carry test and which in case there are time limitations